In [5]:
import pandas as pd
import numpy as np
from pathlib import Path

# Set up paths
PROJECT_ROOT = Path.cwd().parent
PROCESSED_DIR = PROJECT_ROOT / "data" / "processed"

# Load cleaned data
cleaned_file = PROCESSED_DIR / "cleaned_cancer_data.csv"
df = pd.read_csv(cleaned_file)

print(f"Loaded cleaned data: {df.shape[0]:,} rows × {df.shape[1]} columns")
print(f"Columns: {list(df.columns)}")
print("\nFirst few rows:")
print(df.head())

Loaded cleaned data: 6 rows × 5 columns
Columns: ['Year', 'Deaths', 'Population', 'Age-Adjusted Rate', 'Crude Rate']

First few rows:
     Year  Deaths  Population  Age-Adjusted Rate  Crude Rate
0  2018.0  1841.0  81956393.0                2.2         2.2
1  2019.0  1704.0  81679568.0                2.1         2.1
2  2020.0  1702.0  82873432.0                2.0         2.1
3  2021.0  1722.0  82201873.0                2.1         2.1
4  2022.0  1772.0  81903660.0                2.1         2.2


In [6]:
# Year-over-year death trends
print("=== Year-over-Year Death Trends ===")
df_sorted = df.sort_values('Year')

for i in range(1, len(df_sorted)):
    prev_year = df_sorted.iloc[i-1]
    curr_year = df_sorted.iloc[i]
    
    deaths_change = curr_year['Deaths'] - prev_year['Deaths']
    deaths_pct_change = (deaths_change / prev_year['Deaths']) * 100
    
    print(f"{prev_year['Year']} → {curr_year['Year']}:")
    print(f"  Deaths: {prev_year['Deaths']:,} → {curr_year['Deaths']:,} ({deaths_pct_change:+.1f}%)")
    print(f"  Change: {deaths_change:+,} deaths")

=== Year-over-Year Death Trends ===
2018.0 → 2019.0:
  Deaths: 1,841.0 → 1,704.0 (-7.4%)
  Change: -137.0 deaths
2019.0 → 2020.0:
  Deaths: 1,704.0 → 1,702.0 (-0.1%)
  Change: -2.0 deaths
2020.0 → 2021.0:
  Deaths: 1,702.0 → 1,722.0 (+1.2%)
  Change: +20.0 deaths
2021.0 → 2022.0:
  Deaths: 1,722.0 → 1,772.0 (+2.9%)
  Change: +50.0 deaths
2022.0 → 2023.0:
  Deaths: 1,772.0 → 1,829.0 (+3.2%)
  Change: +57.0 deaths


## Year-over-Year Mortality Trends (2018–2023)

Between 2018 and 2019, total deaths declined significantly by 7.4% (−137 deaths), representing the largest single-year decrease in the observed period. Mortality remained relatively stable from 2019 to 2020, with only a slight 0.1% decrease.

Beginning in 2020, deaths increased for three consecutive years. From 2020 to 2021, deaths rose by 1.2%, followed by a 2.9% increase from 2021 to 2022, and a further 3.2% increase from 2022 to 2023. By 2023, total deaths reached 1,829 — the highest level in the dataset.

Overall, the data suggest an initial decline in mortality followed by a sustained upward trend from 2020 onward. Further analysis is needed to determine whether this increase reflects population growth, aging demographics, reporting changes, or broader public health factors.